### ISSUE TITLE
requires "unsafe-eval" Content Security Policy

### ISSUE THREAD

**Issue by TanninOne** [sass/dart-sass#1035]
I'm trying to use dart-sass as a js module from electron, but the module can't be loaded without setting the "unsafe-eval" CSP.
The problematic code is in "tearOffGetter" where it generates a dynamic function with "new Function".

Afaict this code is generated by dart2js and this is a known problem with it, see https://github.com/dart-lang/sdk/issues/37775 The fix seems to be invoking dart2js with --csp.

I assume this has performance implications but maybe you could provide a separate npm package that is built with that flag set?

**Comment by Awjin** [comment]
Interesting. Could you provide more info on your use case? Since Dart Sass is currently a Node.js program, I'm not sure how it would be compatible with WebExtensions to begin with. Does manually invoking `dart2js` with `--csp` solve your problem?

**Comment by TanninOne** [comment]
What I was trying to do was use the module from electron (renderer process) which has the same problem.

I wanted to try the build with dart2js --csp but I'm not familiar with dart. I feel a bit stupid but I didn't manage to build the module and couldn't find where your build setup calls dart2js. I think I managed to build the standalone dart (bin\sass.dart) but when I try to build "dart2js lib\sass.dart" it complains that there is no main function.

**Comment by TanninOne** [comment]
fyi: We're probably going to work around this by using sass from the browser process (which is pure node.js and thus doesn't have the problem) and then use ipc to make the functionality available to the renderer process.

tl;dr is that we don't actually depend on this issue being fixed, I still feel it's something that should be considered though or at least documented.

**Comment by Awjin** [comment]
> tl;dr is that we don't actually depend on this issue being fixed, I still feel it's something that should be considered though or at least documented.

Thanks for the investigation.

> fyi: We're probably going to work around this by using sass from the browser process (which is pure node.js and thus doesn't have the problem) and then use ipc to make the functionality available to the renderer process.

I'll close this issue since there is a reasonable workaround.
